From 7720c9cf3bcba89671d65f05a6d1074fd213cdce Mon Sep 17 00:00:00 2001
From: Daniel Schwierzeck <daniel.schwierzeck@gmail.com>
Date: Sat, 11 Aug 2018 22:37:47 +0200
Subject: [PATCH] MIPS: cache: allow jumps in mips_cache_reset

The $ra register is already saved in $s0. Use $s0 to return from
mips_cache_reset. This allows to use another linked jumps inside
mips_cache_reset.

Signed-off-by: Daniel Schwierzeck <daniel.schwierzeck@gmail.com>
---
 arch/mips/lib/cache_init.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/mips/lib/cache_init.S b/arch/mips/lib/cache_init.S
index b209f23f0a..50caa274b6 100644
--- a/arch/mips/lib/cache_init.S
+++ b/arch/mips/lib/cache_init.S
@@ -421,7 +421,7 @@ l2_unbypass:
 return:
 	/* Ensure all cache operations complete before returning */
 	sync
-	jr	ra
+	jr	R_RETURN
 	END(mips_cache_reset)
 
 /*
